pvlib/solarposition.py

```diff
@@ -1445,3 +1457,118 @@ def sun_rise_set_transit_geometric(times, latitude, longitude, declination,
+    latitude : float
+        Latitude in degrees, positive north of equator, negative to south
+    longitude : float
+        Longitude in degrees, positive east of prime meridian, negative to west
+
+    Returns
+    -------
+    DataFrame
+        The DataFrame will have the following columns:
+        zenith (degrees),
+        elevation (degrees),
+        azimuth (degrees),
+        equation_of_time (seconds),
+        eccentricity,
+        declination (degrees).
+
+    References
+    ----------
+    [1] Spencer (1972) can be found in
+    Sen, Zekai. Solar energy fundamentals and modeling techniques:
+     atmosphere, environment, climate change and renewable energy.
+     Springer Science & Business Media, 2008.
+    """
+
+    julians = datetime2julian(times)
+    julians_2000 = np.asarray(julians, dtype=np.float) - JULIAN_2000
+
+    lat = np.radians(latitude)
+
+    # Compute fractional year (gamma) in radians
+    gamma = 2 * np.pi * (julians_2000 % JULIAN_YEARS) / JULIAN_YEARS
+    cos_gamma = np.cos(gamma), np.cos(gamma * 2), np.cos(gamma * 3)
+    sin_gamma = np.sin(gamma), np.sin(gamma * 2), np.sin(gamma * 3)
+    day_time = (julians_2000 % 1) * 24
+
+    # Eccentricity: correction factor of the earth's orbit.
+    eccentricity = (
+            1.00011 +
+            0.034221 * cos_gamma[0] +
+            0.001280 * sin_gamma[0] +
+            0.000719 * cos_gamma[1] +
+            0.000077 * sin_gamma[1]
+    )
+
+    declination = np.array(declination_spencer71(times.dayofyear))
+
+    # Equation of time (difference between standard time and solar time).
+    eot = (0.000075 + 0.001868 * cos_gamma[0] - 0.032077 * sin_gamma[0]
+           - 0.014615 * cos_gamma[1] - 0.040849 * sin_gamma[1]) * 229.18
+
+    # True local time
+    tlt = (day_time + longitude / 15 + eot / 60) % 24 - 12
+
+    # Solar hour angle
+    ha = np.radians(tlt * 15)
+
+    # Calculate sun elevation.
+    sin_sun_elevation = (np.sin(declination) * np.sin(lat) +
+                         np.cos(declination) * np.cos(lat) * np.cos(ha))
+
+    # Compute the sun's elevation and zenith angle.
+    elevation = np.arcsin(sin_sun_elevation)
+    zenith = np.pi / 2 - elevation
+
+    # Compute the sun's azimuth angle.
+    y = -(np.sin(lat) * np.sin(elevation) - np.sin(declination)) \
+        / (np.cos(lat) * np.cos(elevation))
+    azimuth = np.arccos(y)
+
+    # Convert azimuth angle from 0-pi to 0-2pi.
+    tlt_filter = 0 <= tlt
+    azimuth[tlt_filter] = 2 * np.pi - azimuth[tlt_filter]
+
+    result = pd.DataFrame({'zenith': np.degrees(zenith),
+                           'elevation': np.degrees(elevation),
+                           'azimuth': np.degrees(azimuth),
+                           'eccentricity': eccentricity,
+                           'declination': declination,
+                           'equation_of_time': eot},
+                          index=times)
+    return result
+
+
+def datetime2julian(times):
```

Comment:
Does this function do the same thing as `tools.datetime_to_djd`?

Reply:
I guess it is similar but not the same. djd is limited to days.

Reply:
No not quite the same but it is similar. Actually datetime_to_djd is referencing to the dublin standard and the function is not able to work with arrays. I will have a look If it is possble to adapt my function that it is working with dublin julian days so that we can use my implementation.